google/stackdriver_monitoring/stackdriver_handlers.py

```diff
@@ -13,24 +13,110 @@
 # limitations under the License.
 
 import cgi
+import json
 import logging
 from stackdriver_client import StackdriverClient
 
 
-def accepts_html(request):
+def accepts_content_type(request, content_type):
+  """Determine if the request wants the given content_type as a response."""
   if not hasattr(request, 'headers'):
     return False
   accept = request.headers.get('accept', None)
   if not accept:
     return None
-  return 'text/html' in str(accept)
+  return content_type in str(accept)
+
+
+class BatchProcessor(object):
+  """Helper class for managing events in batch."""
+  def __init__(self, project, stackdriver, data_list,
+               invocation_factory, get_name):
+    """Constructor.
+
+    Args:
+      data_list: [object] The data to operate on.
+    """
+    self.__project = project
+    self.__stackdriver = stackdriver
+    self.__data_list = data_list
+    self.__num_data = len(self.__data_list)
+    self.__invocation_factory = invocation_factory
+    self.__get_name = get_name
+
+    self.batch_response = [None] * self.__num_data
+    self.num_bad = 0  # number of unsuccessful responses.
+    self.num_ok = 0   # number of successful responses.
+    self.was_ok = [None] * self.__num_data
+
+  def handle_batch_response(self, index_str, good, bad):
```

Comment:
The good and bad are actually different types. This is the interface from the batch callback so I cant control their structure. I renamed them.